### DISCUSSION TITLE
`papis export` hangs when piping to clipboard

### DISCUSSION THREAD

**Discussion by ExpandingMan** [papis/papis#953] [Q&A]
If I do just `papis export` in a single line, everything works fine.  However, usually what I want is to pipe that into `xclip` (or whatever).  Bizarrely, if I do `papis export | xclip` it says
```
[INFO] database.cache: Indexing library. This might take a while...
```
and seems to just hang indefinitely.

Is there any way around this?  I'm not even sure how it knows to behaved differently with the pipe.  This happens in seemingly all shells (`nushell`, `fish`, `zsh`).

**Comment by alexfikl** [comment]
```
papers export -f json --all <QUERY> | wl-copy
```
works for me with no hangs. From what you wrote:
* Maybe you can try indexing your library before doing the export (e.g. `papis cache reset` or `papis cache update`)
* A bare `papis export` will open the picker, which likely won't work with piping. Maybe doing a `papis export --all | xclip` works?

**Reply by ExpandingMan** [reply]
Ah, yeah I think that's the core issue, I would like the picker to somehow work with the pipe.  Perhaps an alternative would be to add an option to `export` that allows papis to dump into the clipboard instead of the stdout?  I wonder how people have been doing this.  `papis export --all` *does* work but is usually not what I want.

**Comment by alexfikl** [comment]
@ExpandingMan This might have been a bug that was fixed by #969. The pipe `|` thing just pipes stdout, so I assumed that's why it can't work, but we had a little workaround for that that should be restored now!
